SQL_2709_String_Functions.sql

```diff
@@ -0,0 +1,113 @@
+SELECT RIGHT(EmailAddr, CHARINDEX('@',EmailAddr)) FROM Emails;
+
+--13. Find the position of the first number in the string 'ABC123DEF456'.
+SELECT PATINDEX('%[0-9]%','ABC123DEF456');
+
+--14. Write a query to find all product names from a 'Products' table that contain a number.
+SELECT ProductName FROM Products WHERE PATINDEX('%[0-9]%',ProductName) != 0;
+
+--15. Join the strings 'SQL', 'Server', and '2022' with spaces between them.
+SELECT CONCAT_WS(' ', 'SQL','Server','2022');
+
+--16. From 'Employees' table with 'FirstName' and 'LastName' columns, create a 'FullName' column.
+SELECT CONCAT_WS(' ',FirstName,LastName) AS FullName FROM Employees;
+
+--17. Join the array ('SQL', 'Server', '2022') with a hyphen as the separator.
+SELECT CONCAT_WS('-','SQL', 'Server', '2022');
+
+--18. From an 'Addresses' table, combine 'Street', 'City', 'State', and 'ZIP' columns into a single address string.
+SELECT CONCAT_WS(', ',Street,City,State,ZIP) AS FullAddress from Addresses;
+
+--19. Change all occurrences of 'a' to 'e' in the string 'database management'.
+SELECT REPLACE ('database management','a','e');
+
+--20. From a 'Products' table, write a query to replace all spaces in product names with underscores.
+UPDATE Products SET ProductName = REPLACE (ProductName,' ','_');
+
+--21. Create a string of 10 asterisks (*).
+SELECT REPLICATE('*', 10);
+
+--22. Write a query to pad all product codes in a 'Products' table to a length of 10 characters with leading zeros.
+UPDATE Products SET ProductID = RIGHT((REPLICATE('0',10)+ProductID),10);  
+
+--23. Insert the string 'New ' at the beginning of 'York City'.
+SELECT CONCAT('New', ' York City');
+
+--24. From an 'Emails' table, mask the username part of email addresses, showing only the first and last characters.
+SELECT LEFT(emailid,1) + REPLICATE('*',CHARINDEX('@',emailid)-2) +RIGHT(emailid,len(emailid)-Charindex('@',emailid)+2) FROM Email;
+
+--25. Convert the string 'sql server' to uppercase.
+SELECT UPPER('sql server');
+
+--26. Write a query to convert all customer names in a 'Customers' table to uppercase.
+UPDATE Customers SET CustName = UPPER(CustName);
+
+--27. Convert the string 'SQL SERVER' to lowercase.
+SELECT LOWER ('SQL SERVER');
+
+--28. From a 'Products' table, write a query to convert all product descriptions to lowercase.
+UPDATE Products SET ProdDescription = LOWER(ProdDescription);
+
+--29. Remove trailing spaces from the string 'SQL Server    '.
+SELECT RTRIM('SQL Server    ');
+
+--30. Write a query to remove trailing spaces from all email addresses in an 'Employees' table.
+UPDATE Employees SET email = RTRIM(email);
+
+--31. Remove leading spaces from the string '   SQL Server'.
+SELECT LTRIM('   SQL Server');
+
+--32. From a 'Comments' table, write a query to remove leading spaces from all comment texts.
+UPDATE Comments SET comment = LTRIM(comment);
+
+--33. Display the current date in the format 'dd-MM-yyyy'.
+SELECT FORMAT(GETDATE(), 'dd-ss-yyyy') ;

```

Comment:
Fix the format string for dd-MM-yyyy instead of using ss

Reply:
Changed